infra/templates/az-isolated-service-single-region/ase.tf

```diff
@@ -42,10 +42,17 @@ module "service_plan" {
 
 module "app_service" {
   source                           = "../../modules/providers/azure/app-service"
-  app_service_name                 = var.app_service_name
   service_plan_name                = module.service_plan.service_plan_name
   service_plan_resource_group_name = azurerm_resource_group.admin_rg.name
   app_insights_instrumentation_key = module.app_insights.app_insights_instrumentation_key
+  azure_container_registry_name    = module.container_registry.container_registry_name
+  docker_registry_server_url       = module.container_registry.container_registry_login_server
+  docker_registry_server_username  = module.container_registry.admin_username
+  docker_registry_server_password  = module.container_registry.admin_password
```

Comment:
as mentioned in my earlier note, I'm a bit concerned that we're pulling the acr password from terraform state as opposed to the terraform acr data source reference.

Reply:
I don't think that using a data reference will help us out here as the data reference is also stored in the terraform state.

Here is a trivial example:
```bash
# A test terraform template
$ cat main.tf
terraform {
  backend "azurerm" {
    key = "terraform.tfstate"
  }
}


data "azurerm_container_registry" "acr" {
  name                = "isolatedservicecobaltisolatedtestingacr"
  resource_group_name = "isolated-service-cobalt-isolated-testing-app-rg"
}

# Apply
$ terraform apply
data.azurerm_container_registry.acr: Refreshing state...

Apply complete! Resources: 0 added, 0 changed, 0 destroyed.
```

Now here is a (condensed) version of the state file...
```bash
{
  "resources": [
    {
      "type": "azurerm_container_registry",
      "name": "acr",
      "instances": [
        {
          "attributes": {
            "admin_enabled": true,
            "admin_password": *** CLEAR TEXT PASSWORD REMOVED BY ME ***
            "admin_username": *** CLEAR TEXT USERNAME REMOVED BY ME ***
          }
        }
      ]
    }
  ]
}
```